src/MCPClient/lib/clientScripts/writeDigitalObjectComponentsToArchivesSpace.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python2
+
+from argparse import ArgumentParser
+from csv import DictWriter
+from glob import iglob
+import os
+import sys
+from uuid import uuid4
+
+from main.models import ArchivesSpaceDOComponent
+
+# archivematicaCommon
+from elasticSearchFunctions import getDashboardUUID
+
+from agentarchives.archivesspace import ArchivesSpaceClient
+
+
+def main(current_location, host, port, user, passwd):
+    dashboard_uuid = getDashboardUUID()
+
+    csvfile = open(os.path.join(current_location, 'metadata', 'digital_object_components.csv'), 'w')
+    writer = DictWriter(csvfile, ['component_id', 'digital_object_component_id'])
+    writer.writeheader()
+
+    component_ids = [os.path.basename(path).split('digital_object_component_')[1] for path in
+                     iglob(os.path.join(current_location, 'digital_object_component_*'))]
+
+    components = ArchivesSpaceDOComponent.objects.filter(id__in=component_ids)
+    # exit early to avoid the expense of hitting up ASpace
+    if components.count() == 0:
+        return 0
+
+    client = ArchivesSpaceClient(host=host, port=port, user=user, passwd=passwd)
+
+    # In most circumstances an AIP being processed will be associated
+    # with only a single archival object, but since that isn't guaranteed at
+    # the database level let's operate with the assumption this could happen.
+    archival_object_map = {}
+    for component in components:
+        if component.resourceid not in archival_object_map:
```

Comment:
Tip: `archival_object_map` could be a DefaultDict https://docs.python.org/2/library/collections.html#collections.defaultdict